backend/tests/services/knowledge/structured/test_schema_extractor.py

```
@@ -0,0 +1,187 @@
+        # Boolean values might be detected as BOOLEAN or VARCHAR
+        assert flag_col["type"] in ["BOOLEAN", "BOOL", "VARCHAR", "OBJECT"]
+
+    def test_extract_with_different_delimiters(self):
+        """Test extraction with semicolon delimiter."""
+        # Some CSVs use semicolon as delimiter
+        csv_content = b"name;age;city\nAlice;30;Beijing"
+
+        # The extractor should handle or detect different delimiters
+        # Behavior depends on implementation
+        try:
+            schema = self.extractor.extract_from_csv(csv_content)
```

Ruff (0.15.0)

[error] Local variable `schema` is assigned to but never used

Remove assignment to unused variable `schema`

(F841)